src/plugins/dashboard/public/dashboard_app/no_data/dashboard_app_no_data.tsx

```diff
@@ -44,6 +61,81 @@ export const DashboardAppNoDataPage = ({
       })
     )
   );
+
+  const lensHelpersAsync = useAsync(() => {
+    return lensService?.stateHelperApi() ?? Promise.resolve(null);
+  }, [lensService]);
+
+  useEffect(() => {
+    return () => {
+      abortController?.abort();
+    };
+  }, [abortController]);
+
+  const onTryESQL = useCallback(async () => {
+    abortController?.abort();
+    if (lensHelpersAsync.value && isMounted()) {
```

Comment:
Thanks for adding `isMounted` check. `isMounted` check needs to happen after async functions, so after `getESQLQueryColumns` instead of at the beginning of the function.

Reply:
Thinking about this more, maybe the isMounted check is not even needed since `getESQLQueryColumns` request is canceled on unmount.

Reply:
yes I wanted to tell you the same tbh, I am removing this